rest_framework/serializers.py

```diff
@@ -388,10 +400,7 @@ def to_internal_value(self, data):
 
         ret = OrderedDict()
         errors = OrderedDict()
-        fields = [
-            field for field in self.fields.values()
-            if (not field.read_only) or (field.default is not empty)
-        ]
+        fields = self._to_internal_value_fields
```

Comment:
Using a local variable provides faster access in Python. https://www.python.org/doc/essays/list2str/